praw/objector.py

```diff
@@ -66,6 +85,35 @@ def objectify(self, data):
                 .issubset(set(data.keys()))):
             parser = self.parsers['ModmailConversation']
             return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'actionTypeId' in data.keys():
```

Comment:
The reason why I used `{'date', 'id', 'name'}.issubset(set(data.keys())` for mapping is mainly because I wanted those matchers to be stronger. In the ones that you've added, this parsing breaks if, for instance, ModmailAction adds an `isAdmin` field.

I realize the`elif isinstance(data, dict) and 'user' in data` is also pretty vague, though that was at a time where there were very few such checks. Now that this is being leveraged to parse more I think we need to have stronger matching.